mirrord/agent/src/main.rs

```diff
@@ -285,12 +290,20 @@ impl ClientConnectionHandler {
                         break;
                     }
                 },
-                message = self.tcp_sniffer_api.recv(), if sniffer_available => {
+                // use to prevent closing when sniffer is not available
+                // while avoiding possible busy loop (infinite iteration of None)
+                message = async {
+                    if let Some(ref mut sniffer_api) = self.tcp_sniffer_api {
+                        sniffer_api.recv().await
+                    } else {
+                        unreachable!()
```

Comment:
Is this unreachable even though the async expression is evaluated (but not polled) even when the precondition does not hold?

Reply:
I copied it from other codebase tbh, but pretty sure this should work since sniffer api shouldn't become unavailable during the run, only at load so this should be okay